src/Forms/FormScaffolder.php

```diff
@@ -35,32 +35,26 @@ class FormScaffolder
     public bool $mainTabOnly = false;
 
     /**
-     * @var boolean $ajaxSafe
-     * @deprecated 5.3.0 Will be removed without equivalent functionality.
+     * Array of field names to use as an allow list.
+     * If left blank, all fields from {@link DataObject->db()} will be included unless explicitly ignored.
      */
-    public $ajaxSafe = false;
-
-    /**
-     * @var array $restrictFields Numeric array of a field name whitelist.
-     * If left blank, all fields from {@link DataObject->db()} will be included.
-     */
-    public $restrictFields;
+    public array $restrictFields = [];
 
     /**
      * Numeric array of field names and has_one relations to explicitly not scaffold.
      */
     public array $ignoreFields = [];
 
     /**
-     * @var array $fieldClasses Optional mapping of fieldnames to subclasses of {@link FormField}.
+     * Optional mapping of fieldnames to subclasses of {@link FormField}.
      * By default the scaffolder will determine the field instance by {@link DBField::scaffoldFormField()}.
      */
-    public $fieldClasses;
+    public array $fieldClasses = [];
 
     /**
-     * @var boolean $includeRelations Include has_many and many_many relations
+     * Include has_many and many_many relations
      */
-    public $includeRelations = false;
+    public bool|array $includeRelations = false;
```

Comment:
Looking at the code that uses this, you could use it to explicitly only allow `has_many` relations but not `many_many` relations (or vice versa) - in which case it needs to be an associative array. Hence `bool|array`.